esx_service/utils/auth.py

```diff
@@ -0,0 +1,234 @@
+import random
+import logging
+import auth_data
+import os
+
+auth_mgr = None
+def connect_auth_db():
+    global auth_mgr
+    if not auth_mgr:
+        auth_mgr = auth_data.AuthorizationDataManager()
+        auth_mgr.connect()
+       
+def find_tenant_by_vm(vm_uuid):
+    """
+        Find tenant which owns this VM by querying the auth DB
+    """
+    global auth_mgr
+    cur = auth_mgr.conn.execute(
+                "select * from vms where vm_id=?",
+                (vm_uuid,)
+    )
+    tenant_uuid = None
+    tenant_name = None
+    result = cur.fetchone()
+    if result:
+        tenant_uuid = result['tenant_id']
+        cur = auth_mgr.conn.execute(
+                "select * from tenants where id=?",
+                (tenant_uuid,)
+        )
+        result = cur.fetchone()
+        if result:
+            tenant_name = result['name']
+            logging.debug("Found tenant_uuid %s, tenant_name", tenant_uuid, tenant_name)
+
+    return tenant_uuid, tenant_name
+
+def find_privileges_by_tenant_and_datastore(tenant_uuid, datastore):
+    """
+        Return privileges for given (tenant_uuid, datastore) pair by
+        querying the auth DB 
+    """
+    global auth_mgr
+    privileges = []
+    logging.debug("find_privileges tenant_uuid=%s datastore=%s", tenant_uuid, datastore)
+    cur = auth_mgr.conn.execute(
+                "select * from privileges where tenant_id=? and datastore=?",
+                (tenant_uuid,datastore)    
+    )
+    privileges = cur.fetchone()
+    return privileges
+
+def has_mount_privilege(privileges):
+    logging.debug(privileges)
+    if not privileges:
+        return False
+    logging.debug("mount_volume=%d", privileges['mount_volume'])
+    return privileges['mount_volume']
+
+def has_create_privilege(privileges):
+    logging.debug(privileges)
+    if not privileges:
+        return False
+    logging.debug("create_volume=%d", privileges['create_volume'])
+    return privileges['create_volume']   
+
+def has_delete_privilege(privileges):
+    logging.debug(privileges)
+    if not privileges:
+        return False
+    logging.debug("delete_volume=%d", privileges['delete_volume'])
+    return privileges['delete_volume']
+
+"""
+    For a given size string, return values in MB.
+    Example:
+    '100MB': return 100
+    '100GB': return 100*1024
+"""
+def convert_to_MB(vol_size_str):
+    unit = vol_size_str[-2:]
+    value = int(vol_size_str[:-2])
+    conversions = {'MB' : 1,
+                   'GB' : 1024,
+                   'TB' : 1024*1024,
+                   'PB' : 1024*1024*1024,
+    }
+    
+    if unit.upper() in conversions.keys():
+        value = value*conversions[unit]
+    else:
+        logging.error("Invalid volume size")
+    return value  
+
+def get_vol_size(opts):
+    if not opts:
```

Comment:
Done. Make the change to check both opts is empty and opts does not have key "size".